docs/source/guides/contributor/chapters/plugins.rst

```diff
@@ -27,8 +27,65 @@ world output in the console.
           :py:data:`avocado.core.output.LOG_JOB`, which can be used to
           output to job log files when running a job.
 
+Registering configuration options (settings)
+============================================
+
+It is very common for a plugin to allow users to configure something. A plugin
```

Comment:
Acting like the bad advisor today :D 

"It is usual for a plugin to allow users to do some degree of configuration based on arguments/parameters."

Reply:
Doing with some changes.